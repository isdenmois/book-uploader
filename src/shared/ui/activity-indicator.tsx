import React, { FC } from 'react'
import { ActivityIndicator as RNActivityIndicator, ActivityIndicatorProps } from 'react-native'
import { Theme, useTheme } from './theme'

type Props = ActivityIndicatorProps & {
  color: keyof Theme['colors']
}

export const ActivityIndicator: FC<Props> = ({ color, ...props }) => {
  const { colors } = useTheme()

  return <RNActivityIndicator {...props} color={colors[color]} />
}
