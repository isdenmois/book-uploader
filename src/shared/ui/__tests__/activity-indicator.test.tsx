import React from 'react'
import { render } from '@testing-library/react-native'
import { ActivityIndicator } from '../activity-indicator'

test('ActivityIndicator', () => {
  expect(render(<ActivityIndicator color='primary' />).toJSON()).toMatchSnapshot()
})
