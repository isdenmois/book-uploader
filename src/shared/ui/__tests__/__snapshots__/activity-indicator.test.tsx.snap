// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActivityIndicator 1`] = `
<ActivityIndicator
  animating={true}
  color="#999999"
  hidesWhenStopped={true}
  size="small"
/>
`;
